Sources/Filters/Sources/ArrowSource/index.js

```diff
@@ -0,0 +1,119 @@
+import macro              from 'vtk.js/Sources/macro';
+import vtkAppendPolyData  from 'vtk.js/Sources/Filters/General/AppendPolyData';
+import vtkConeSource      from 'vtk.js/Sources/Filters/Sources/ConeSource';
+import vtkCylinderSource  from 'vtk.js/Sources/Filters/Sources/CylinderSource';
+import vtkMatrixBuilder   from 'vtk.js/Sources/Common/Core/MatrixBuilder';
+
+// ----------------------------------------------------------------------------
+// vtkArrowSource methods
+// ----------------------------------------------------------------------------
+
+function vtkArrowSource(publicAPI, model) {
+  // Set our className
+  model.classHierarchy.push('vtkArrowSource');
+
+  function requestData(inData, outData) {
+    if (model.deleted) {
+      return;
+    }
+
+    const cylinder = vtkCylinderSource.newInstance({ capping: true });
+    cylinder.setResolution(model.shaftResolution);
+    cylinder.setRadius(model.shaftRadius);
+    cylinder.setHeight(1.0 - model.tipLength);
+    cylinder.setCenter(0, (1.0 - model.tipLength) * 0.5, 0.0);
+
+    const cylinderPD = cylinder.getOutputData();
+    const cylinderPts = cylinderPD.getPoints().getData();
+    const cylinderNormals = cylinderPD.getPointData().getNormals().getData();
+
+    // Apply transformation to the cylinder
+    vtkMatrixBuilder
+      .buildFromDegree()
+      .rotateZ(-90)
+      .apply(cylinderPts);
+    vtkMatrixBuilder
+      .buildFromDegree()
+      .rotateZ(-90)
+      .apply(cylinderNormals);
+
```

Comment:
Why don't you just build 1 matrix and apply it on `cylinderPts` then on `cylinderNormals`?

Reply:
We may also update `vtkMatrixBuilder` to make sure `apply` also return `this` so you could also write it that way if you want:

```
  vtkMatrixBuilder
       .buildFromDegree()
       .rotateZ(-90)
       .apply(cylinderPts);
       .apply(cylinderNormals);
```